bin/hveto

```diff
@@ -307,6 +307,13 @@ else:
        pchannel, livetime)
     logger.critical(message)
 
+# cluster primary triggers
+clusterkwargs = cp.getparams('primary', 'cluster-')
+if clusterkwargs:
+    primary = primary.cluster(**clusterkwargs)
+    logger.info("%d primary events remain after clustering over %s" %
+                (len(primary), clusterkwargs['rank']))
```

Comment:
@tjma12, there should probably be some logic that either sets a default for `clusterrkwargs['rank']`, or requires the user to pass it if clustering is enabled. I personally prefer the former option, e.g. after line 312 you could simply add this:

```python
    clusterkwargs.setdefault('rank', 'snr')
```

or similar.

Reply:
Are you concerned about the logger throwing an error? All of the arguments for `EventTable.cluster()` are currently set as required arguments rather than kwargs with defaults, so if any of the arguments, including `clusterkwargs['rank']`, aren't provided the clustering will fail and throw an error saying that a rank is required. I've kinda handed off error handling to the `EventTable.cluster()` method in this case.

Reply:
@tjma12, indeed, that is my concern. For documentation purposes, I believe you're referring to [**this line**](https://github.com/gwpy/gwpy/blob/master/gwpy/table/table.py#L710) of the `gwpy.table` module, in which case I agree, this will throw a more appropriate error before the logger.